Sources/MapboxCoreNavigation/ReplayLocationManager.swift

```diff
@@ -53,7 +53,7 @@ open class ReplayLocationManager: NavigationLocationManager {
      A handler that is called when `ReplayLocationManager` finished replaying `locations` and about to start from the
      beginning. Return false to stop replaying `locations`.
      */
-    var onReplayLoopCompleted: ((ReplayLocationManager) -> Bool)?
+    public var onReplayLoopCompleted: ((ReplayLocationManager) -> Bool)?
```

Comment:
It would be a bit less awkward to expose a protocol for this, so the developer doesn’t have to worry as much about creating retain cycles by accident.

Is there precedence for the `on` naming convention in Swift? It feels like a Java-ism. SwiftUI has methods named this way, but that’s a bit different. I think `completionHandler` or `originReturnHandler` would be a more conventional name.

Reply:
> It would be a bit less awkward to expose a protocol for this, so the developer doesn’t have to worry as much about creating retain cycles by accident.

There are two sides to this:
- Delegates are less likely to produce retain-cycles (although you can still do it, and I've seen this many times).
- Block-based API is more enjoyable to use and has better logic locality. 

An alternative approach can be exposing a configuration struct of some kind that will take care of retain-cycles. Apple also provides more and more block-based APIs:
- https://developer.apple.com/documentation/uikit/uiaction
- https://developer.apple.com/documentation/uikit/uicollectionview/cellregistration

At the same time, NavSDK tends to use protocols for everything. That's why a delegate pattern can be preferable. 

> Is there precedence for the on naming convention in Swift? It feels like a Java-ism. SwiftUI has methods named this way, but that’s a bit different. I think completionHandler or originReturnHandler would be a more conventional name.

Maybe it is just me, but it doesn't sound like Java-ism to me. "On" is a good indicator of the event: something has happened, and this block will be called. The idea is taken from the delegate patter where you start with the object that caused the event and have an event like parameter preposition, for example:
- `func tableView(_ tableView: UITableView, shouldBeginMultipleSelectionInteractionAt indexPath: IndexPath) -> Bool` => `should`
- `func tableView(UITableView, didBeginMultipleSelectionInteractionAt: IndexPath)` => `did`

Anyway, I don't have objections to renaming to something else or migrating to the delegate pattern. I chose block-based API here because it is less awkward to use in tests, and we already see that this class is used in unit tests by others as well: https://github.com/ably/ably-asset-tracking-swift/blob/2acaf27b3ea98aa6700650f2623849e285ee1487/Tests/SystemTests/PublisherAuthenticationSystemTests.swift#L124


We are in alpha, so we can make breaking changes in alpha releases.

Reply:
Renamed to `replayCompletionHandler` in https://github.com/mapbox/mapbox-navigation-ios/pull/3550

Reply:
> The idea is taken from the delegate patter where you start with the object that caused the event and have an event like parameter preposition

Yes, this is a good pattern to follow, it’s just that “on” is a different pattern altogether. The equivalent would be “didCompleteLoop”.